File: Server/src/MatchMaking.cpp
#include "MatchMaking.hpp"

MatchMaking::MatchMaking() {}

MatchMaking::~MatchMaking() {}

Match* MatchMaking::CreateMatch(int IDMatch)
{
    // # Armazena o ponteiro para os dois jogadores que serão pareados.
    auto PID1 = PlayerList[0];
    auto PID2 = PlayerList[1];

    // # Remove os dois jogadores da lista de espera.
    PlayerList.erase(PlayerList.begin(), PlayerList.begin() + 2);

    return new Match(IDMatch, PID1, PID2);
}

File: Server/matchmaking.mjs
export class MatchMaking
{
    playerList = [];
    match_id_counter = 0;

    AddPlayerToQueue(player)
    {
        this.playerList.push(player);
    }

    IsMatchFound()
    {
        return this.playerList.length > 1;
    }

    CreateMatch()
    {
        let PID1 = this.playerList[0];
        let PID2 = this.playerList[1];

        // # Remove os dois jogadores da lista de espera.
        this.playerList.splice(0, 2);

        let IDMatch = ++this.match_id_counter;

        return {"IDMatch": IDMatch, "PID1": PID1, "PID2": PID2};
    }
}

